panels/models/attraction.py

```diff
@@ -33,6 +34,30 @@ def sluggify(s):
     return RE_SLUG.sub('-', s).lower().strip('-')
 
 
+def treeify(items, keys, value_key=None):
+    if not keys:
+        return items
+    keys = listify(keys)
+    last_key = keys[-1]
+    treeified = OrderedDict()
+    for item in items:
+        current = treeified
+        for key in keys:
+            attr = key(item) if callable(key) else getattr(item, key)
+            if attr not in current:
+                current[attr] = [] if key is last_key else OrderedDict()
```

Comment:
If I'm reading this all correctly, there's only one set of branching points, right? Everything else linearly digs down. So the data structure looks like:

```  ---
 / | \
/  |  \
   |   \
        \
```

is that correct?

Reply:
I'll add some docs, but this function can nest arbitrarily deep.